src/test/functional/mutationServices.test/savedItemsMutationService-upsert.integration.ts

```diff
@@ -601,9 +575,44 @@ describe('UpsertSavedItem Mutation', () => {
         });
         expect(eventTracker.callCount).to.equal(0);
       });
+
+      //this test passes.
```

Comment:
given we already have tests to add this item, we could just spy on the writeDb and check that it called using the server that works (non-null db).

Reply:
nice idea! looking into if its possible to spyOn object that gets a parameter.

tried something like this to make sure writeDb is called

```
  const db = writeClient(); //passed to apollo server
  const writeDbSelectSpy = jest.spyOn(db('list'), 'select');
  expect(writeDbSelectSpy.mock.calls.length).to.be.greaterThan(1);
```

The call `jest.toBeCalled() and jest.toHaveBeenCalled()` doesn't work with our test framework..
https://codewithhugo.com/jest-fn-spyon-stub-mock/

Reply:
Jest isn't great, but sinon pretty much always works :)

Reply:
I agree that we should either remove this commented out test or fix.

Reply:
sinon also have a similar behavior, 

```
 const spy = await sinon.spy(db('list'), 'insert');
 expect(spy.called).greaterThan(0);
 spy.resetHistory();
```

so tried spying on `insert` or `select` or `queryContext` and the assertions all failed with 0 calls.

https://github.com/knex/knex/issues/3775 might be related to this issue, we don't have `context` as they described,

i am also confused by why the test commented out would make the following test suite fail.